stdlib/LibGit2/src/remote.jl

```diff
@@ -157,9 +159,11 @@ julia> LibGit2.push_url(LibGit2.get(LibGit2.GitRemote, repo, "origin"))
 """
 function push_url(rmt::GitRemote)
     ensure_initialized()
-    url_ptr = ccall((:git_remote_pushurl, libgit2), Cstring, (Ptr{Cvoid},), rmt.ptr)
-    url_ptr == C_NULL && return ""
-    return unsafe_string(url_ptr)
+    GC.@preserve rmt begin
```

Comment:
ditto this and below

Reply:
Idem.

Reply:
For all these similar cases, I propose to remove the `.ptr` annotation and add a comment to explain why the `GC.@preserve` is needed.

Reply:
:+1: